compiler/test/dotty/tools/dotc/reporting/ErrorMessagesTests.scala

```diff
@@ -1511,4 +1511,22 @@ class ErrorMessagesTests extends ErrorMessagesTest {
       assertEquals(tailRegMessages, Set("variable", "value", "object", "class"))
     }
 
+  @Test def notAnExtractor() =
+    checkMessagesAfter(FrontEnd.name) {
+      """
+        | class Foo(val name: String)
+        | object Test {
+        |   def runFoo() = {
+        |     new Foo("foo") match {
+        |       case Foo(name) => println(name)
+        |       case _ => println("Dunno")
+        |     }
+        |   }
+        | }
+      """.stripMargin
+    }.expect { (ictx, messages) =>
+      implicit val ctx: Context = ictx
+      assertMessageCount(2, messages)
+      assertTrue(messages(1).isInstanceOf[NotAnExtractor])
```

Comment:
We also try to test the content of the error message. In your case, you could do something like
```scala
val NotAnExtractor(tree) = messages.head
assertEqual("Foo", tree.show)
```